kolibri/plugins/user/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.deprecation import MiddlewareMixin
+
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import KolibriAnonymousUser
+
+ALLOWED_PATH_LIST = [
```

Comment:
This middleware should be in core, not in the user plugin (the setup wizard is an anti-pattern in this respect, and not to be copied).

We should also not have an explicit list of paths that are allowed, as this means that no other plugin can ever exempt itself from the guest access restriction.

Instead, we should use a view decorator to exclude it from this middleware (see, for example, the CSRF exempt decorator in Django).